packages/experience-builder-sdk/src/core/componentRegistry.ts

```diff
@@ -129,3 +129,35 @@ export const getComponentRegistration = (id: string) => {
   }
   return componentRegistry.get(id);
 };
+
+export const addComponentRegistration = (componentRegistration: ComponentRegistration) => {
+  componentRegistry.set(componentRegistration.definition.id, componentRegistration);
+};
+
+export const createDesignComponentRegistration = ({
+  definitionId,
+  defintionName,
+  component,
+}: {
+  definitionId: string;
+  defintionName?: string;
+  component: ComponentRegistration['component'];
+}) => {
+  const componentRegistration = componentRegistry.get(definitionId);
+
+  if (componentRegistration) {
+    return componentRegistration;
```

Comment:
would it make sense to align the response shape to what is on line 162?

Reply:
Align the response shape in what way? This check is for incase we have the definition of that component saved already, then it returns it instead of re-setting it. The shape is already aligned

Reply:
On line 62 we return both - a component and a component Definition.

Reply:
If you check what's stored in the component registry, it's the component and its definition. So 162 and this line have the same response shape